modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/PartitionCountersNeighborcastFuture.java

```diff
@@ -111,15 +112,22 @@ public void init() {
                 cctx.io().send(n, new PartitionCountersNeighborcastRequest(cntrs, futId, tx.topologyVersion()), SYSTEM_POOL);
             }
             catch (IgniteCheckedException e) {
-                if (!(e instanceof ClusterTopologyCheckedException))
-                    log.warning("Failed to send partition counters to remote node [node=" + peer + ']', e);
-                else
+                if (e instanceof ClusterTopologyCheckedException)
                     logNodeLeft(peer);
+                else
+                    log.warning("Failed to send partition counters to remote node [node=" + peer + ", txId=" + tx.nearXidVersion() + ']', e);
 
                 miniFut.onDone();
             }
         }
 
+        if (log.isInfoEnabled()) {
+            log.info("Starting delivery partition countres to remote nodes [txId=" + tx.nearXidVersion() +
+                ", futId=" + futId +
+                ", topVer=" + tx.topologyVersion() +
+                ", msgs=" + peerMsgs);
```

Comment:
I'm a bit worried about this message's length. In the test it's ok, but can it become too big on the client's clusters where there are much more nodes and partitions?

Reply:
The length of `peerMsgs` list equals to the size of nodes participating in transaction, and the sizes of this list (i.e. PartitionUpdateCountersMessage ) is limited with partition counters associated with this particular transaction. Transaction details we already log in cases like LRT, so I dont think this change can cause some additional problems.